extensions/markdown/src/extension.ts

```diff
@@ -0,0 +1,170 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+'use strict';
+
+import * as vscode from 'vscode';
+import * as path from 'path';
+import { ExtensionContext, TextDocumentContentProvider, EventEmitter, Event, Uri, ViewColumn } from "vscode";
+
+const hljs = require('highlight.js');
+const mdnh = require('markdown-it-named-headers');
+const md = require('markdown-it')({
+	html: true,
+	highlight: function (str, lang) {
+		if (lang && hljs.getLanguage(lang)) {
+			try {
+				return `<pre class="hljs"><code><div>${hljs.highlight(lang, str, true).value}</div></code></pre>`;
+			} catch (error) { }
+		}
+		return `<pre class="hljs"><code><div>${md.utils.escapeHtml(str)}</div></code></pre>`;
+	}
+}).use(mdnh, {});
```

Comment:
@cbreeden your request is valid to me. People always need to customize the Markdown rendering process to meet their need. A final perfect solution is allowing users to load markdown-it anomynous extensions on the fly. But I don't find this feature documented on their website. So personaly I suggest:

1. Find out or add `on-the-fly` loading mechanism for markdown-it. Maybe create a feature request on their repo.
2. Emebed this feature into our seperate Markdown Extension. Provide users a configuration (an array is simple enough) to inject their markdown-it extensions written in JS.

Since we've already moved the markdown stuff out of VS Code, we may want to keep all related features in markdown or its dependencies instead of here.

If you are willing to help file these issues/requests to corresponding repositories, it would be really awesome. Otherwise I can do it myself.